src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -170,7 +172,9 @@ public static unsafe bool IsSortable(string text)
         [OnDeserializing]
         private void OnDeserializing(StreamingContext ctx)
         {
-            m_name = null;
+            // TODO-NULLABLE: this becomes null for a brief moment before deserialization
+            //                after serialization is finished it is never null
+            m_name = null!;
```

Comment:
Can we just do it on the field declaration and leave this body empty?

Reply:
I think this may possibly change the logic - I don't think de-serialization can happen on existing objects but technically it is possible to do it (not sure why original code was assigning in here instead of field initialization so just in case there is some special case I prefer to leave it out of this changeset)

Reply:
Ok, sounds good.